telegram-bot/main.go

```diff
@@ -252,6 +258,14 @@ func main() {
 				fetchnote(ID, update.Message.Text, *client)
 			case "paste":
 				paste(ID, update.Message)
+			case "me":
+				user := update.Message.From.UserName
+				msg := update.Message.Text
+				s := strings.SplitN(msg, " ", 2)
+				if len(s) == 2 {
+					me(user, ID, s[1])
+					bot.DeleteMessage(tbot.NewDeleteMessage(ID, update.Message.MessageID))
+				}
```

Comment:
The entire code should be abstracted to the `me` function instead of just `bot.Send`

Reply:
I couldn't put the bot delete command inside the function, I don't think the user update text would be defined inside the function, if there is a way, do tell. I have committed the changes of what I could do.

Reply:
Pass the whole message interface and you should be able to access all the underlying functions.

Reply:
It is done, marking as resolved.